### ISSUE TITLE
Breaks with PageForms 4.1.1

### ISSUE THREAD

**Issue by Rudloff** [SemanticMediaWiki/SemanticFormsSelect#64]
Hello,

I'm not sure if it is a PF bug or a SFS bug.
I'm using SFS 2.1.0. When I update PF to 4.1.1, the `Special:FormEdit` page crashes with this error on forms that use SFS:
```
Fatal error: Class '\SFS\SemanticFormsSelect::init' not found in /home/pierre/www/archi-mediawiki/extensions/PageForms/includes/PF_FormPrinter.php on line 1615
```

If I revert back to 4.1, I don't get this error anymore.

**Comment by yaronkoren** [comment]
Yes - the way in which classes are loaded was changed; or rather, there were two ways to load form input classes, a preferred way and a deprecated way, and the deprecated way was finally removed. @gesinn-it had a fix for this - he mentioned it here:

https://github.com/SemanticMediaWiki/SemanticFormsSelect/issues/62

...but I guess he hasn't checked it in.

**Comment by gesinn-it** [comment]
you can try out our current fork of SFS. This is a major refactoring but not yet ready for merge. We are already working with this internally. Works great. @yaronkoren would be cool to get the patches for PF merged that we discussed recently. With these changes, SFS would also support "mapping properties"

**Comment by kghbln** [comment]
> you can try out our current fork of SFS. 

[I guess you are referring to this one](https://github.com/gesinn-it/SemanticFormsSelect)

> Yes - the way in which classes are loaded was changed

Self reminder: PF does not follow semver.

**Comment by yaronkoren** [comment]
Thanks for reminding yourself!

**Comment by kghbln** [comment]
The reason is that I somewhere have an instance using Composer for PF. Do not have the time to change this now so every edit to this issue will remind me to change this.

**Comment by mwjames** [comment]
> Thanks for reminding yourself!

For people who are not familiar with semver (and given the current
situation it is fairly evident), http://semver.org/ simple states:

MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards-compatible manner, and
PATCH version when you make backwards-compatible bug fixes.

Sometimes incompatible API changes are unavoidable but that is why one
should provide a flag to retain functionality until a MAJOR version
allows for it to be removed and this is what can be called responsible
development everything else is just hacking and makes deploying of
software unprofessional.

Cheers

**Comment by piaf666** [comment]
bug at line 1617.
`PHP Fatal error:  Class '\\SFS\\SemanticFormsSelect::init' not found in /opt/mediawiki/extensions/PageForms/includes/PF_FormPrinter.php on line 1617`
Configuration :
MediaWiki: 1.27.3
PHP: 5.6.30 (apache2handler)
SFS_VERSION : 2.1.0-alpha

Cheers

**Comment by Megajoul** [comment]
Hello,
Do somebody still work on this bug? Can we expect a new version of SF Select?
Cheers

**Comment by mwjames** [comment]
> Do somebody still work on this bug? Can we expect a new version of SF
> Select?

If someone fixes the aforementioned issues then yes, otherwise no.

**Comment by tyndale-onomastics** [comment]
Would anybody be willing to explain how I could modify the code for my own installation, to make SF Select work? I'm not sufficiently familiar with php to be able to tell from the discussion.

With thanks,
Elizabeth

**Comment by kghbln** [comment]
I believe this is now resolved with the 3.0.0 release of this extension which is afaik at least working with Page Forms 4.3.1
